src/server.c

```diff
@@ -4201,9 +4203,29 @@ int processCommand(client *c) {
 
 /* ====================== Error lookup and execution ===================== */
 
+/* Users who abuse lua error_reply will generate a new error object on each
+ * error call, which can make server.errors get bigger and bigger. This will
+ * cause the server to block when calling INFO (we also return errorstats by
+ * default). To prevent the damage it can cause, when a misuse is detected,
+ * we will print the warning log and disable the errorstats to avoid adding
+ * more new errors. It can be re-enabled via CONFIG RESETSTAT. */
+#define ERROR_STATS_NUMBER 1000
 void incrementErrorCount(const char *fullerr, size_t namelen) {
+    /* errorstats is disabled, return ASAP. */
+    if (!server.errors_enabled) return;
+
     void *result;
     if (!raxFind(server.errors,(unsigned char*)fullerr,namelen,&result)) {
+        if (server.errors->numele >= ERROR_STATS_NUMBER) {
+            serverLog(LL_WARNING,
+                      "Errorstats stopped adding new errors because the number of "
+                      "errors reached the limit, may be misuse of lua error_reply, "
+                      "please check INFO ERRORSTATS, this can be re-enabled via "
+                      "CONFIG RESETSTAT.");
```

Comment:
i think we wanna print the contents of `server.errors` to the log when disabling them.
and then release that memory `server.errors`, maybe replacing it with a single element that states it was disabled.
this way the user has a chance to find the abusing script (by looking at the log).
in the current form, the user may not notice it got disabled.

Reply:
if 1000 is too much, i don't mind reducing the limit to 128

Reply:
I did print errors before, by using genRedisInfoString, but the output was too much and was truncated, so I changed it to this way.

now i change to a new way, here is the log output:
```
89253:M 18 Mar 2024 16:56:08.886 # Errorstats stopped adding new errors because the number of errors reached the limit, may be misuse of lua error_reply, please check INFO ERRORSTATS, this can be re-enabled via CONFIG RESETSTAT.
89253:M 18 Mar 2024 16:56:08.886 # Current errors code list: 1, 10, 100, 101, 102, 103, 104, 105, 106, 107, 108, 109, 11, 110, 111, 112, 113, 114, 115, 116, 117, 118, 119, 12, 120, 121, 122, 123, 124, 125, 126, 127, 128, 13, 14, 15, 16, 17, 18, 19, 2, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 3, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 4, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 5, 50, 51, 52, 53, 54, 55, 56, 57, 58, 59, 6, 60, 61, 62, 63, 64, 65, 66, 67, 68, 69, 7, 70, 71, 72, 73, 74, 75, 76, 77, 78, 79, 8, 80, 81, 82, 83, 84, 85, 86, 87, 88, 89, 9, 90, 91, 92, 93, 94, 95, 96, 97, 98, 99
```

info errorstats output:
```
# Errorstats
errorstat_ERRORSTATS_DISABLED:count=1
```